plugins/module_utils/linode_common_list.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+"""This module allows users to list SSH keys in their Linode profile."""
+
+from __future__ import absolute_import, division, print_function
+
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional
+
+from ansible_collections.linode.cloud.plugins.module_utils.linode_common import (
+    LinodeModuleBase,
+)
+from ansible_collections.linode.cloud.plugins.module_utils.linode_docs import (
+    global_authors,
+    global_requirements,
+)
+from ansible_collections.linode.cloud.plugins.module_utils.linode_helper import (
+    construct_api_filter,
+    get_all_paginated,
+)
+from ansible_specdoc.objects import (
+    FieldType,
+    SpecDocMeta,
+    SpecField,
+    SpecReturnValue,
+)
+
+spec_filter = {
+    "name": SpecField(
+        type=FieldType.string,
+        required=True,
+        description=[
+            "The name of the field to filter on.",
+        ],
+    ),
+    "values": SpecField(
+        type=FieldType.list,
+        element_type=FieldType.string,
+        required=True,
+        description=[
+            "A list of values to allow for this field.",
+            "Fields will pass this filter if at least one of these values matches.",
+        ],
+    ),
+}
+
+
+@dataclass
+class ListModuleParent:
+    """
+    Represents a single parent resource ID for a list module.
+    This is intended to be used for nested resources (e.g. Instance Config)
+    """
+
+    field: str
+    display_name: str
+    type: FieldType
+
+
+class ListModuleBase(LinodeModuleBase):
+    """A common module for listing API resources given a set of filters."""
+
+    display_name = ""
+    result_field = ""
+    endpoint = ""
+    docs_url = ""
+    parents: List[ListModuleParent] = []
+    examples = []
+    response_samples = []
+
+    def __init__(self) -> None:
+        self.module_arg_spec = self.spec.ansible_spec
+        self.results: Dict[str, Any] = {self.result_field: []}
+
+        super().__init__(module_arg_spec=self.module_arg_spec)
```

Comment:
I think we can initialize the attributes in `__init__()` function and let the sub-classes call the parent's `__init__()`. I feel by this way, we would align with the typical object-oriented design pattern.

Reply:
```python
class Parent:
    a: int
    b: str
    def __init__(self, a, b):
        self.a = a
        self.b = b

class Child1(Parent):
    def __init__(self, a, b):
        super().__init__(a, b)

class Child2(Parent):
    def __init__(self):
        super().__init__(123, "abc")
```

Just an example to clarify my thought.

Another advantage I can think of is it will let the developer of the sub-class be aware of all the attributes and ensure each attribute is considered, and people may be confusing about how the attributes work in the declarative sub-class.

And feel free to resolve this is we are not planning it in this PR.

Reply:
Ahh I see, thanks for the feedback! We can discuss this further in the standalone PR.